### PR TITLE
feature/documentation (new 'make docs' target)

### PR DESCRIPTION
**Pull request by abelardojarab**
* this is the new branch with 'make docs' target.

requires following packages to be installed: doxygen

and from python:
requests==2.18.4
docutils==0.14
pygments==2.2.0
sphinx==1.6.5
sphinx-rtd-theme
sphinxcontrib-pandoc-markdown
breathe
travis-sphinx

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,9 +19,14 @@ jobs:
       script: scripts/test-build.sh
     - # BAT
       script: scripts/test-gtapi-mock-drv.sh
+    - # BAT
+      script: scripts/build-documentation.sh
```
Comment (abelardojarab) [inline .travis.yml]: rename stage do not put with the BAT

File: .travis.yml [hunk 2]
```diff
@@ -7,7 +7,8 @@ install:
   - pip install --user pylint
   - pip install --user pycodestyle
   - pip install --user jsonschema
-
+gemfile:
+    - Gemfile
 jobs:
   include:
     - stage: BAT
```
Comment (abelardojarab) [inline .travis.yml]: rename as stage1, for example.

File: doc/CMakeLists.txt [hunk 3]
```diff
@@ -32,3 +32,8 @@ find_package(Doxygen)
 if(DOXYGEN_FOUND)
   add_subdirectory(doxygen)
 endif(DOXYGEN_FOUND)
+
```
Comment (abelardojarab) [inline doc/CMakeLists.txt]: rename targets: 
* doc as 'make doxygen'
* sphinx_html as 'make docs'

File: doc/sphinx/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  DEPENDS doc)
+
+add_custom_target(copy-markdown-files ALL
+  COMMAND cmake -E copy_directory ${CMAKE_SOURCE_DIR}/doc/src ${BINARY_BUILD_DIR}/docs)
+
+add_custom_target(copy-include-files ALL
+  COMMAND cmake -E copy_directory ${CMAKE_SOURCE_DIR}/common/include ${BINARY_BUILD_DIR}/docs/fpga_api/include
+  DEPENDS copy-markdown-files)
+
+add_custom_target(copy-sphinx-contrib-files ALL
```
Comment (abelardojarab) [inline doc/sphinx/CMakeLists.txt]: we are installing all sphinx extensions from pip, remove this.

File: doc/sphinx/conf.py.in [hunk 5]
```diff
@@ -5,47 +5,39 @@
...
+import sphinx_fontawesome
+
+# Pandoc-based markdown parser for sphinx
+from sphinxcontrib.pandoc_markdown import MarkdownParser
 
 # Enabled extensions
 extensions = [
     'sphinx.ext.autodoc',
-    'sphinx.ext.intersphinx',
+    # 'sphinx.ext.intersphinx',
```
Comment (abelardojarab) [inline doc/sphinx/conf.py.in]: remove comments if not planned to be used in future.

**Comment by southerngs** [comment]
Does this build with Travis CI?  I looked at the output but I didn't see where the actual docs get built.  I think it would be useful to have this as a CI target to make sure it's working.

Do we have a list of packages needed for building on CentOS?

File: doc/sphinx/conf.py.in [hunk 6]
```diff
@@ -67,9 +57,10 @@
 # The version info for the project you're documenting
 #
 # The short X.Y version.
-version = u'0.9.0'
+version = u'@INTEL_FPGA_API_VERSION@'
```
Comment (r-rojo) [inline doc/sphinx/conf.py.in]: Should this be `@INTEL_FPGA_API_VERSION_MAJOR@.@INEL_FPGA_API_VERSION_MINOR@` ?

**Pull request merged**